### DISCUSSION TITLE
db error fatal remaining connection slots are reserved for non-replication superuser connections

### DISCUSSION THREAD

**Discussion by ramezj** [prisma/prisma#22508] [Q&A]
### Question

I'm getting this error whenever I try to work on my project : **db error fatal remaining connection slots are reserved for non-replication superuser connections**

this is how I connect to the database : 

```js
import { PrismaClient } from '@prisma/client'
let prisma;
//check if we are running in production mode
if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient()
} else {
//check if there is already a connection to the database
  if (!global.prisma) {
    global.prisma = new PrismaClient()
  }
  prisma = global.prisma
}

export { prisma };
```

and then i import prisma from that file and use it.

### How to reproduce (optional)

- Sign up for the Supabase free tier
- Create a database
- Connect to the database using the same code I'm connecting with.

### Expected behavior (optional)

Obviously to not get this error.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Windows
- Database: PostgreSQL running on Supabase ( free tier )
- Node.js version: v20.9.0
- Prisma version : 5.7.1

**Comment by janpio** [comment]
When do you get this error? Instantly? 
Can you connect with other database clients?

**Comment by namhihi237** [comment]
Error in connector: Error querying the database: db error: FATAL: remaining connection slots are reserved for non-replication superuser connections

Sometime I get this error,

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
